neuralforecast/models/tft.py

```diff
@@ -127,56 +123,37 @@ def __init__(self, hidden_size,
                 setattr(self, attr+'_bias', None)
 
     def _apply_embedding(self,
-                         cat: Optional[Tensor],
                          cont: Optional[Tensor],
-                         cat_emb: Iterable[Any], 
                          cont_emb: Tensor,
                          cont_bias: Tensor,
                          ):
 
-        if (cat is None) and (cont is None):
-            return None
-
-        if (cat is not None):
-            e_cat = torch.stack([embed(cat[...,i]) \
-                               for i, embed in enumerate(cat_emb)], dim=-2)
         if (cont is not None):
             #the line below is equivalent to following einsums
             #e_cont = torch.einsum('btf,fh->bthf', cont, cont_emb)
             #e_cont = torch.einsum('bf,fh->bhf', cont, cont_emb)          
             e_cont = torch.mul(cont.unsqueeze(-1), cont_emb)
             e_cont = e_cont + cont_bias
-
-        if (cat is not None) and (cont is None):
-            return e_cat
-
-        if (cat is None) and (cont is not None):
             return e_cont
-
-        if (cat is not None) and (cont is not None):
-            return torch.cat([e_cat, e_cont], dim=-2)
         
         return None
 
-    def forward(self, target_inp, s_cat_inp=None, s_cont_inp=None, k_cat_inp=None, k_cont_inp=None, o_cat_inp=None, o_cont_inp=None):
+    def forward(self, target_inp, 
+                stat_exog=None, futr_exog=None, hist_exog=None):
         # temporal/static categorical/continuous known/observed input 
         # tries to get input, if fails returns None
 
         # Static inputs are expected to be equal for all timesteps
         # For memory efficiency there is no assert statement
-        s_cat_inp = s_cat_inp[:,0,:] if s_cat_inp is not None else None
-        s_cont_inp = s_cont_inp[:,0,:] if s_cont_inp is not None else None
+        stat_exog = stat_exog[:,:] if stat_exog is not None else None
 
-        s_inp = self._apply_embedding(s_cat_inp, s_cont_inp,
-                                      cat_emb=self.s_cat_embed,
+        s_inp = self._apply_embedding(stat_exog,
```

Comment:
Let's protect this calls by naming the inputs.
In future revisions of the code we might change the order of the inputs and we can make this robust.